Upload files

JoelForamitti opened the pull request:
# Type of change

<!-- Please delete options that are not relevant. -->
- [x] New feature (non-breaking change which adds functionality)

# Description

- Upload function for Item API with multipart form

# Checklist:

- [x] I have commented my code (or ChatGPT did), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings, neither in my IDE nor in my browser
- [x] I have added tests that prove my fix is effective or that my feature works

nanu-c commented on app/handlers/router.go:
```diff
@@ -85,8 +85,8 @@ func GetRouter() (r *chi.Mux) {
 	r.Handle("/img/*", http.StripPrefix("/img/", fs))
-	fs = http.FileServer(http.Dir("docs"))
-	r.Handle("/docs/*", http.StripPrefix("/docs/", fs))
+	fs2 := http.FileServer(http.Dir("docs"))
+	r.Handle("/docs/*", http.StripPrefix("/docs/", fs2))
```
maybe rename `fs2` to `fs_docs`

nanu-c commented on app/handlers/handlers_test.go:
```diff
@@ -65,11 +67,63 @@ func TestUsers(t *testing.T) {
+	err := json.Unmarshal(res.Body.Bytes(), &resItems)
+	utils.CheckError(t, err)
+	require.Equal(t, 1, len(resItems))
+	require.Equal(t, "Test item", resItems[0].Name)
+
+	// Update (multipart form!)
+	body := new(bytes.Buffer)
+    writer := multipart.NewWriter(body)
+    writer.WriteField("Name", "Updated item name")
+    writer.WriteField("nonexistingfieldname", "10")
+    image, _ := writer.CreateFormFile("Image", "test.jpg")
+    image.Write([]byte(`i am the content of a jpg file :D`))
+    writer.Close()
+	utils.TestRequestMultiPart(t, r, "PUT", "/api/items/"+itemID+"/", body, writer.FormDataContentType(), 200)
+
+	// Read
```
Please run the linter to align the the lines nicely ;)

JoelForamitti replied:
done

JoelForamitti commented on app/handlers/handlers.go:
```diff
@@ -212,43 +261,43 @@ func CreateItem(w http.ResponseWriter, r *http.Request) {
+		utils.ErrorJSON(w, errors.New("invalid form"), http.StatusBadRequest)
+		return
+	}
+	log.Info("mForm: ", mForm)
```
removed

nanu-c commented on app/handlers/router.go:
```diff
@@ -83,10 +83,10 @@ func GetRouter() (r *chi.Mux) {
-	fs = http.FileServer(http.Dir("docs"))
-	r.Handle("/docs/*", http.StripPrefix("/docs/", fs))
+	fs_img := http.FileServer(http.Dir("img"))
+	r.Handle("/img/*", http.StripPrefix("/img/", fs_img))
```
How do you avoid to handle urls like `../../../etc/passwd`?

JoelForamitti replied:
I tested it and it is not possible. The only way that going to parent that is allowed is /../doc since it points to another fileserver.

JoelForamitti commented:
I changed docs to doc to avoid overlap with the /docs directory where the documentation lies. I also added volumes for file persistence for production
